Test.c

```diff
@@ -4,11 +4,166 @@
+                {  1,1,1,1,1,1,1,0 }  , { 2,0,0,0,0,0,0,0 } };
+   // Keys and expected index positions
+   int search[][2] = { { 3 , 2 } , { 0 , 5 }, { 801 , 4 } , { -9 , 3 },
+                       { 1 , 0 } , { 5 , -1 } };
+   int length = sizeof (search) / sizeof (search[0]);
+   printf (YELLOW"\n                                    Test BinarySearch()\n"RESET);
+   PrintLine ();
+   printf ("              Sorted Input               |   Key   |");
+   printf ("  Found at  |     Status     |   Search    \n");
+   PrintLine ();
+   for (int i = 0; i < length; i++) {
+      int lenOfArr = sizeof (arr[i]) / sizeof (arr[i][0]);
+      PrintArray (arr[i], lenOfArr);
+      int index = BinarySearch (arr[i], lenOfArr, search[i][0]);
+      printf (" |   %3d   |    %4d    ", search[i][0], index);
+      printf ("|    %10s   ", index == -1 ? BLUE"Not found"RESET : CYAN"  Found  "RESET);
+      printf ("|    %6s    \n", search[i][1] == index ? CYAN"Pass"RESET : MAGENTA"Fail"RESET);
+      PrintLine ();
+   }
+}
+
+static void TestSort () {
+   int arr[][MAX_SIZE] = { { 1,2,3,5,4,6,7,8 } , { 8,7,5,4,3,1 } ,
+                  { -1,-3,-5,-9,-234,-90,-123,0 }, { 0,5,-567,67,8,2,9,124 } ,
+                  { 1,1,1,1,1,1,1 } , { 2 } };
+   int length = sizeof (arr) / sizeof (arr[0]);
+   printf (YELLOW"\n                                    Test HeapSort()\n"RESET);
+   PrintLine ();
+   printf ("                  Input                  |                   Output        ");
+   printf ("          |  Sort  \n");
+   for (int i = 0; i < length; i++) {
+      PrintLine ();
+      int lenOfArr = sizeof (arr[i]) / sizeof (arr[i][0]);
+      PrintArray (arr[i], lenOfArr);
+      HeapSort (arr[i], lenOfArr);
+      printf (" | ");
+      PrintArray (arr[i], lenOfArr);
+      printf ("  |  %6s  \n",
+              IsSorted (arr[i], lenOfArr) ? CYAN"Pass"RESET : MAGENTA"Fail"RESET);
+   }
+   PrintLine ();
+}
+
+/// <summary>Gets user input and returns true if 'y' is pressed.</summary>
+static bool Choice () {
+   while (1) {
+      switch (getchar ()) {
+         case 'y': return true;
+         case 'n': return false;
+         case '\n': break;
+         default: while (getchar () != '\n'); break;
+      }
+      printf ("Please enter a valid option. \nEnter (y/n) ");
+   }
+}
+
+/// <summary>Gets integer from the user.</summary>
+static int GetInt (char* input) {
+   while (1) {
+      printf ("%s : ", input);
+      // NewLine character as a string to compare with strpbrk() function to check for '\n'
+      char newLineChar[2] = { '\n','\0' };
+      char strInput[15], * endptr = NULL;
+      fgets (strInput, 15, stdin);
+      // Converts input string to long long int
+      long long int num = strtoll (strInput, &endptr, 10);
+      // If the string length exceeds 14, clear input buffer
+      if (strpbrk (strInput, newLineChar) == NULL) while (getchar () != '\n');
+      if (*endptr != '\n' || num > INT_MAX || num < INT_MIN || strInput[0] == '\n') {

```

Comment:
FYI: in this particular case you can also compare with errno == ERANGE and it would work in a similar fashion.